src/curl.c

```diff
@@ -180,28 +180,28 @@ curl_multi_sock_cb(CURL *easy, curl_socket_t fd, int what, void *envp,
 {
 	(void) easy;
 	struct curl_env *env = (struct curl_env *) envp;
-	struct ev_io *watcher = (struct ev_io *) watcherp;
+	env->watcher = (struct ev_io *)watcherp;
 
 	if (what == CURL_POLL_REMOVE) {
 		say_debug("curl %p: remove fd=%d", env, fd);
-		assert(watcher != NULL);
-		ev_io_stop(loop(), watcher);
+		assert(env->watcher != NULL);
+		ev_io_stop(loop(), env->watcher);
 		++env->stat.sockets_deleted;
-		mempool_free(&env->sock_pool, watcher);
+		mempool_free(&env->sock_pool, env->watcher);
 		return 0;
 	}
 
-	if (watcher == NULL) {
-		watcher = mempool_alloc(&env->sock_pool);
-		if (watcher == NULL) {
-			diag_set(OutOfMemory, sizeof(*watcher), "mempool",
+	if (env->watcher == NULL) {
+		env->watcher = mempool_alloc(&env->sock_pool);
+		if (env->watcher == NULL) {
+			diag_set(OutOfMemory, sizeof(*env->watcher), "mempool",
 				 "curl sock");
 			return -1;
 		}
-		ev_io_init(watcher, curl_sock_cb, fd, 0);
-		watcher->data = env;
+		ev_io_init(env->watcher, curl_sock_cb, fd, 0);
+		env->watcher->data = env;
 		++env->stat.sockets_added;
-		curl_multi_assign(env->multi, fd, watcher);
+		curl_multi_assign(env->multi, fd, env->watcher);
 		say_debug("curl %p: add fd=%d", env, fd);
 	}
 
```

Comment:
nit: isn't it better to keep this code above as it was and only here assign `env->watcher = watcher`?